server/task.go

```diff
@@ -696,10 +696,14 @@ func (s *Server) patchTaskStatus(ctx context.Context, task *api.Task, taskStatus
 		taskStatusPatch.Result = &resultStr
 	}
 
-	taskPatched, err := s.store.PatchTaskStatus(ctx, taskStatusPatch)
+	taskPatchedList, err := s.store.PatchTaskStatus(ctx, taskStatusPatch)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to change task %v(%v) status", task.ID, task.Name)
 	}
+	if len(taskPatchedList) != 1 {
```

Comment:
What could lead to this case? Add a comment, or do the validation before we call s.store.PatchTaskStatus()?